sdb/locator.py

```diff
@@ -57,33 +57,27 @@ def caller(self, objs: Iterable[drgn.Object]) -> Iterable[drgn.Object]:
         """
 
         out_type = self.prog.type(self.output_type)
+        baked = dict()
+        for (_, method) in inspect.getmembers(self, inspect.ismethod):
+            if not hasattr(method, "input_typename_handled"):
+                continue
+            baked[sdb.type_canonicalize_name(
+                self.prog, method.input_typename_handled)] = method
```

Comment:
I wonder if in the future we can/want to make this global so we don't have to recreate it every time - kinda like `all_commands` where we add new entries as Classes get registered. Similarly for walkers and pretty printers

Reply:
Unfortunately I don't think we can do that as the subclasses are registered, because `sdb.prog` isn't set up yet at that point.  We could do something to cache it instead, but I figured I'd leave that for when we notice it being an actual performance problem.